### ISSUE TITLE
[1.10.2] Issues

### ISSUE THREAD

**Issue by lehjr** [MachineMuse/MachineMusePowersuits#756]
These are the currently known issues (will be updated as things change )in the current build:
**Some future release**:
- Look into replacing opengl calls with GLStateManager calls. 
- Revisit block breaking code and cleanup/optimize. Also need to check and make sure things are actually working properly.
- Change the precision of sliders to allow for better rounding of values. Current setup is notoriously horrible when trying to get a nice round number. ( I think I can do it, started it, will go back to it) see: https://github.com/MachineMuse/MachineMusePowersuits/issues/758
- look into adding a "fortune upgrade" to the powerfist. see here: https://github.com/MachineMuse/MachineMusePowersuits/issues/759
- look into silk touch upgrade: https://github.com/MachineMuse/MachineMusePowersuits/issues/797
- thermal generator module install cost needs to be fixed
- Alternate Tinker table. The iconic model had to be butchered to deal with rendering glitches. To make up for it, I would like to move the model to a JSON type so that players can replace it with one of their own from resource packs. 
- particle trails because why not. This may be put on hold until rendering issues can be worked out. https://github.com/MachineMuse/MachineMusePowersuits/issues/654
- wireless terminal modules for Refined storage (want to rewrite a little bit): https://github.com/MachineMuse/MachineMusePowersuits/issues/782
 - wireless terminal module for AE2: clean up code a little. 
**Postponed for now**
- Elytra flight will be on hold for a bit. It's a bit too complicated to deal with at the moment. There's a pr in progress for forge that will make things easier. Waiting to see how that goes first. Other solutions are possible, but terrible.

**Comment by Desempregago** [comment]
Ah, and fix the speed boost (the actual speed doesn't even change, doesn't matter the value set in the tinker table). Tool modules (shears, leaf blower and lux capacitor) are crashing the game when used.

**Comment by lehjr** [comment]
Just telling me things are crashing is meaningless without logs

**Comment by lehjr** [comment]
"Ah, and fix the speed boost " You're gonna have to be a lot more specific than that.

**Comment by Desempregago** [comment]
The speed boost doesn't change, actually it works even if I uninstall the module (salvage)

**Comment by Desempregago** [comment]
About the lux capacitor crash

Description: Tesselating block model

java.lang.IndexOutOfBoundsException: Index: 1, Size: 1
	at java.util.ArrayList.rangeCheck(Unknown Source)
	at java.util.ArrayList.get(Unknown Source)
	at net.minecraft.client.renderer.vertex.VertexFormat.func_177344_b(SourceFile:90)
	at net.minecraft.client.renderer.VertexBuffer.func_178962_a(VertexBuffer.java:408)
	at net.minecraft.client.renderer.BlockModelRenderer.renderQuadsFlat(BlockModelRenderer.java:390)
	at net.minecraft.client.renderer.BlockModelRenderer.func_187497_c(BlockModelRenderer.java:154)
	at net.minecraftforge.client.model.pipeline.ForgeBlockModelRenderer.func_187497_c(ForgeBlockModelRenderer.java:86)
	at net.minecraft.client.renderer.BlockModelRenderer.func_187493_a(BlockModelRenderer.java:72)
	at net.minecraft.client.renderer.BlockModelRenderer.func_178267_a(BlockModelRenderer.java:52)
	at net.machinemuse.powersuits.client.render.entity.EntityRendererLuxCapacitorEntity.doRender(EntityRendererLuxCapacitorEntity.java:49)
	at net.machinemuse.powersuits.client.render.entity.EntityRendererLuxCapacitorEntity.func_76986_a(EntityRendererLuxCapacitorEntity.java:24)
	at net.minecraft.client.renderer.entity.RenderManager.func_188391_a(RenderManager.java:398)
	at net.minecraft.client.renderer.entity.RenderManager.func_188388_a(RenderManager.java:375)
	at net.minecraft.client.renderer.RenderGlobal.func_180446_a(RenderGlobal.java:811)
	at net.minecraft.client.renderer.EntityRenderer.func_175068_a(EntityRenderer.java:1741)
	at net.minecraft.client.renderer.EntityRenderer.func_78471_a(EntityRenderer.java:1566)
	at net.minecraft.client.renderer.EntityRenderer.func_181560_a(EntityRenderer.java:1345)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1076)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:310)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:394)
	at org.multimc.EntryPoint.listen(EntryPoint.java:170)
	at org.multimc.EntryPoint.main(EntryPoint.java:54)

**Comment by Desempregago** [comment]
Shears and any other "tool" module crash
Description: Unexpected error

java.lang.ClassCastException: net.minecraft.item.ItemShears cannot be cast to net.minecraft.item.ItemTool
	at net.machinemuse.powersuits.powermodule.tool.ToolHelpers.isEffectiveTool(ToolHelpers.java:34)
	at net.machinemuse.powersuits.powermodule.tool.ShearsModule.canHarvestBlock(ShearsModule.java:119)
	at net.machinemuse.powersuits.event.HarvestEventHandler.handleBreakSpeed(HarvestEventHandler.java:36)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler_486_HarvestEventHandler_handleBreakSpeed_BreakSpeed.invoke(.dynamic)
	at net.minecraftforge.fml.common.eventhandler.ASMEventHandler.invoke(ASMEventHandler.java:90)
	at net.minecraftforge.fml.common.eventhandler.EventBus.post(EventBus.java:185)
	at net.minecraftforge.event.ForgeEventFactory.getBreakSpeed(ForgeEventFactory.java:172)
	at net.minecraft.entity.player.EntityPlayer.getDigSpeed(EntityPlayer.java:859)
	at net.minecraftforge.common.ForgeHooks.blockStrength(ForgeHooks.java:206)
	at net.minecraft.block.Block.func_180647_a(Block.java:532)
	at net.minecraft.block.state.BlockStateContainer$StateImplementation.func_185903_a(BlockStateContainer.java:399)
	at net.minecraft.client.multiplayer.PlayerControllerMP.func_180511_b(PlayerControllerMP.java:240)
	at PlayerControllerOF.func_180511_b(PlayerControllerOF.java:43)
	at net.minecraft.client.Minecraft.func_147116_af(Minecraft.java:1459)
	at net.minecraft.client.Minecraft.func_184117_aA(Minecraft.java:2179)
	at net.minecraft.client.Minecraft.func_184118_az(Minecraft.java:1961)
	at net.minecraft.client.Minecraft.func_71407_l(Minecraft.java:1749)
	at net.minecraft.client.Minecraft.func_71411_J(Minecraft.java:1055)
	at net.minecraft.client.Minecraft.func_99999_d(Minecraft.java:371)
	at net.minecraft.client.main.Main.main(SourceFile:124)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at net.minecraft.launchwrapper.Launch.launch(Launch.java:135)
	at net.minecraft.launchwrapper.Launch.main(Launch.java:28)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source)
	at java.lang.reflect.Method.invoke(Unknown Source)
	at org.multimc.onesix.OneSixLauncher.launchWithMainClass(OneSixLauncher.java:310)
	at org.multimc.onesix.OneSixLauncher.launch(OneSixLauncher.java:394)
	at org.multimc.EntryPoint.listen(EntryPoint.java:170)
	at org.multimc.EntryPoint.main(EntryPoint.java:54)

**Comment by lehjr** [comment]
which "speed boost" are you referring to

**Comment by lehjr** [comment]
and which version are you using?

**Comment by Desempregago** [comment]
By speed boost, I mean the leg module, and I'm using the latest 1.10.2 version (Alpha 005)

**Comment by lehjr** [comment]
There should be no problem with that. I was just using it fine a few minutes ago.

**Comment by Desempregago** [comment]
But it's weird that the speed boost is still being applied even after module removal

**Comment by eyeonus** [comment]
Also mentioned was that tinkering it to have more or less effect doesn't apparently work. It's the same amount of speed boost.

I haven't tested this myself.

**Comment by Desempregago** [comment]
/\ This.

**Comment by lehjr** [comment]
I just tried it again myself, even recorded it. I am not seeing what you are describing at all. 
Edit: video here: https://vimeo.com/208369919

**Comment by lehjr** [comment]
If I can't reproduce it, I can't fix it.

**Comment by Desempregago** [comment]
Weird... On my client, the game crashes with a Tesselating Model exception when I shoot a Lux Capacitor 

And the speed boost doesn't change at all, doesn't matter the level or setting

**Comment by lehjr** [comment]
What mods do you have installed?

**Comment by Desempregago** [comment]
[Modlist](http://imgur.com/xPWK83F)

**Comment by lehjr** [comment]
Interesting. That's a long list and will be time consuming to replicate. Do you have Industrialcraft installed? And are you using anything like Optifine?

**Comment by lehjr** [comment]
wait, I do see optifine. That might be the cause of the crash for the lux capaicitor. I'll have to try it out later.

**Comment by Desempregago** [comment]
I'm not using IC2, OptiFine U7 (latest)

**Comment by lehjr** [comment]
Well, I was able to replicate the issue with the speed assist module with IC2 removed. I'll have to spend some time on that to figure it out.

**Comment by lehjr** [comment]
Testing confirmed that the lux capacitor crash is caused by Optifine. Unfortunately, a fix will be time consuming due to the inability to use optifine in a development environment.

**Comment by eyeonus** [comment]
This is why I just don't use optifine.

**Comment by lehjr** [comment]
Agreed, but I'll still try and fix it anyway. I didn't find any other issues with optifine, but I didn't do much testing either. The code that's causing issues is likely to change anyway because the animation is horrible. Instead of a smooth flight, the lux capacitor zig zags all over until it lands.

**Comment by lehjr** [comment]
OK, so I have a fix for the optifine crash, and I think for the tool crash as well. The sprint assist module will need more work. I'm not entirely sure of cause of the issue yet.

**Comment by lehjr** [comment]
New alpha is up

**Comment by lehjr** [comment]
@Fogrits I'll be looking at the sprint assist module today and with any luck, the fix should be in the next alpha.

**Comment by lehjr** [comment]
Still working on that sprint assist, but no one noticed the swimming boost didn't actually boost your speed in the water? :P

**Comment by eyeonus** [comment]
Haven't gone in the water yet. Still working on getting the jet pack modules installed in my game. Don't have much time on workdays.

**Comment by lehjr** [comment]
Probably gonna rewrite the whole speed boost module. The current setup uses a weird attribute modifier setup that causes the game to use a bunch of tooltips "when equipped in head slot... when equipped in chest slot.... ". Also fixing it so the speed boost is only applied when the player is actually wearing the armor legs instead of just holding them.

**Comment by lehjr** [comment]
Messing with Eytra style flight just using NBT data (no access transformers or reflection) in a LivingUpdateEvent handler. It's kinda glitchy, especially on turns:  https://vimeo.com/209986481

**Comment by lehjr** [comment]
So I was experimenting with the Elytra stuff. It's interesting but kinda glitchy:

**Comment by lehjr** [comment]
Lightning summoner will be fixed in the next alpha

**Comment by Desempregago** [comment]
The Elytra "module" will be compatible with the jetpack module?
By compatible, I mean using the jetpack as a "jet propeller", like in 1.11.

**Comment by lehjr** [comment]
the Elytra stuff needs some ironing out. One of the issues is the style of flight compared to vanilla, and the mapping of controls. MPS already has a controllable glider which uses shift and forward by default, but the jetpack doesn't really give it any extra thrust, and the animation doesn't rotate the player. Plus there's the issue of wings themselves. The 3d armor has the jetpack model, but the skinned armor only has the skins. I might be able to use the vanilla Elytra models and animation for the skinned armor, but it would look horrible for the 3d armor.

**Comment by lehjr** [comment]
Looks like in order to use the wavefront models I have for the power fist, I will have to create new textures. I was hoping to avoid it so texture packs wouldn't need to be overhauld, but despite what some people claim, you actually do need square textures. I'll include the UV mapping template. Not much else I can do. Armor textures are still OK though. Lux Capacitor changed awhile ago. Most of the textures did get moved around but otherwise are unchanged. 
```
[23:05:08] [Client thread/ERROR]: -------------------------
[23:05:08] [Client thread/ERROR]:     The following other errors were reported for domain powersuits:
[23:05:08] [Client thread/ERROR]: -------------------------
[23:05:08] [Client thread/ERROR]:     Problem: broken aspect ratio and not an animation
[23:05:08] [Client thread/ERROR]:       textures/models/tool.png
```

**Comment by lehjr** [comment]
so the texture for the powerfist will just be the same uv layout just double the height

**Comment by EzerArch** [comment]
Will "elytra style" be just for looks? I think of giving it a functioning purpose: to reduce [drag](https://en.wikipedia.org/wiki/Drag_(physics)), so to say.

A [jetpack-only flight](https://imgur.com/joAeotW) would have a lower top speed and less mobility than the [elytra style](https://www.youtube.com/watch?v=xPoSqmWK_Os). It would require the glider module because, you know, wings are necessary to glide and sustain a cruise flight. :P

**Comment by eyeonus** [comment]
How does elytra flight work anyway?

From what I've seen it's like how Batman does it in the Arkham series, but
I always have flight by the time I get to the End.

**Comment by eyeonus** [comment]
I'm not sure if this is an issue or intended operation of the cooling tank, but I have no heat, ever, on my suit.

Also, traveling to the Nether drops my suit power from 4.01 to 2.60 instantly, and I don't have mechanical assistance module or the dimensional tear module installed.

Traveling back probably does as well, but, wireless recharger.

**Comment by lehjr** [comment]
Heat mechanics will return in the next alpha. They were turned off early in the porting when I was still getting familiar with the internal Minecraft changes. All of that is updated in the code just waiting for the next alpha. Next alpha will probably be released today or tomorrow, whenever I finish implementing the new power fist models. The models themselves are done now, but I need to work on the scale, rotation, position and the code that switches them when firing.

**Comment by lehjr** [comment]
The mechanics and even the operation of the elytra flight are still in conceptual design stages, meaning that nothing is really ironed out. I've been working on pretty much everything else in preparation for the next alpha and there hasn't been any time to even think about the elytra stuff.

**Comment by eyeonus** [comment]
I meant vanilla, not the power suit elytra flight mode.

**Comment by lehjr** [comment]
I haven't even had much of a chance to experiment with the vanilla elytra. It seems pretty limited from what I can tell though.

**Comment by eyeonus** [comment]
I'll try to get one when I go to the end and find out for myself, I suppose.

**Comment by lehjr** [comment]
damned vanilla bugs aren't helping. Probably doesn't help when LWJGL doesn't even mark depreciated functions as depreciated because it might be inconvenient. So despite forge trying to push support for latest builds, vanilla code is still built on outdated lwjgl.

**Comment by lehjr** [comment]
just had to file 1 vanilla bug report and add a "me too" to another

**Comment by lehjr** [comment]
OK, so I fixed the rendering glitch related to the "glow" settings for rendering (affects power fist, armor, and Tinker Table). 
Among other things, glowOn() disables standard item lighting, but glowOff() does not enable it again. So I fixed that. I also need to look at some of the GL calls for the power fist and Tinker Table and ditch anything not needed. The thing is, I really don't know if the pros of going to the new powerfist models outweigh the cons. With the rendering glitches fixed, there really doesn't seem to be much incentive, especially if it breaks texture packs more than I've already broken them with directory structure changes. Dunno, maybe I'll push changing over the models to 1.11.2 and focus on the rest of the issues.

**Comment by lehjr** [comment]
Looks like coloring the power fist with the old models is going to be an issue. The color bleeds into other model base models nearby. So in order to use the Tint modules for the power fist, there is no choice but to use the new models I made. 
![2017-04-05_19 09 33](https://cloud.githubusercontent.com/assets/1084367/24731191/6c059f42-1a36-11e7-88a7-e55b2423fa24.png)

**Comment by lehjr** [comment]
WTH? handleItemState is supposed to be passing the itemstack instance equipped in the main hand, but my testing reveals that stackIn is instead the first inventory **type match**, numerically by slot index.
This means if you have 2 of the same type, no matter how different they are, only the first one is returned. This explains why I was having issues and could not pinpoint the source. At least the entity is correct.

**Comment by lehjr** [comment]
Looks like I'll be changing the third person view a bit. Now the power fist will be moved toward the player more and out to the side more, so that less of the view is covered, matching the current held item rendering better: 
![2017-04-08_14 08 49](https://cloud.githubusercontent.com/assets/1084367/24831281/1af13b9e-1c65-11e7-990c-29e2e91cc5ba.png)

**Comment by lehjr** [comment]
Power Fist with tint module installed
![2017-04-08_20 46 55](https://cloud.githubusercontent.com/assets/1084367/24833671/bd21a7a8-1c9e-11e7-8b69-c9dc57d1f3b1.png)

**Comment by lehjr** [comment]
armor coloring coming along
![2017-04-09_09 47 08](https://cloud.githubusercontent.com/assets/1084367/24837784/b8f1bd10-1d09-11e7-98bf-e597777cc2f2.png)

**Comment by lehjr** [comment]
Plasma charge is now a meter since there isn't any reliable way to render the plasma bolt while charging.

**Comment by lehjr** [comment]
Plasma bolt charge meter (also the meters are now more dynamic). The power meter is no longer blue in order to better differentiate it from the water meter:
![2017-04-10_14 44 46](https://cloud.githubusercontent.com/assets/1084367/24877179/58a256f6-1dfc-11e7-9fb3-2f2ebbbb1387.png)

**Comment by EzerArch** [comment]
> Plasma bolt charge meter (also the meters are now more dynamic).

Just to clarify: is this an energy meter of its own or something else? I thought the plasma bolt consumed the energy directly from the batteries.

If it is a cool-down meter of some sort, MC 1.9+ renders a cool-down bar next to the crosshair (I made a custom one for my resource pack). Not sure if it can be used for MPS.

![Image](https://1.bp.blogspot.com/-TgR2k8PE3I4/WM328-yyzJI/AAAAAAAAA7Y/ynyewwiZPAQ05NsGvkXQUChaQo5orqzSQCEw/s1600/2017-03-19_00.06.52.png)

**Comment by lehjr** [comment]
The plasma bolt originally had a charge animation where it would show the bolt growing in front of the power fist before release. It was done inside the model rendering, but unfortunately, doing this with the new models is almost impossible due to the rendering being handled by Minecraft itself. So a meter had to be created to show the charge.

**Comment by lehjr** [comment]
Remember, the plasma bolt does more damage the more it is charged before release. As to why the need for the new models, the old models did cause some rendering glitches, and they could not be colored without causing the color to bleed out into other models, like chests. Eventually, the Tinker table model will be also be replaced. In the mean time, I might make the portable unit fully functional.

**Comment by lehjr** [comment]
when I get that "transparent armor" module set up, I should probably do that for the power fist too, but should I make it still visible in first person view?

**Comment by eyeonus** [comment]
Yes, but see-through?

Could you replicate the growing thing with a particle effect?

I've seen some pretty crazy particle effects. Just look at Blood Magic.

**Comment by eyeonus** [comment]
Also, could you maybe make the bars horizontal and lined up with their numeric equivalents?

(EDIT: The graphical meters)

**Comment by lehjr** [comment]
See through probably, but particle effects I would like to stay away from. Keep in mind that the rendering loop is outside the model, so there is no way to render that. The old model had the rendering loop as part of the model, but with threaded rendering, it's unreliable, and the mode GL calls, the more likely something will break. Unfortunately, I need to keep the code simple enough to trouble shoot and maintain, and only complex enough to do what it needs to do.

**Comment by eyeonus** [comment]
No worries. I figured a particle effect would be an easy fix, since it wouldn't be part of the fist model, but I have no idea how graphics work in this game, so I'm not surprised if it isn't worth the hassle.

What say horizontal?

**Comment by lehjr** [comment]
The thing is, the meter is only needed when charging up the plasma bolt, so it's not going to be seen as much as everything else. On top of that, since my time is limited, I have to weigh the time spent adding new features against time spent getting the mod polished enough so that people can actually use it (eternal Alpha stage is not an option). New features take both time to develop and time to fix/polish, time that could be spent fixing things or adding things like the Elytra flight.

**Comment by eyeonus** [comment]
I'm not asking for a new feature wrt the meter displays. I'm asking for an alteration of an existing feature to made it more user friendly.

Instead of
![current vertical meters](https://cloud.githubusercontent.com/assets/1733600/24929972/0df41738-1ec5-11e7-894c-01ef480fdc68.png)
Have:
![proposed horizontal meters](https://cloud.githubusercontent.com/assets/1733600/24929973/11b62992-1ec5-11e7-8088-1e9dc0670d8f.png)

This way the numerical values are lined up with their graphical meters.

**Comment by lehjr** [comment]
It's a new feature if it requires drastic code changes. I simply don't have
time for that at the moment

**Comment by eyeonus** [comment]
Where's the code that draws the meters?

**Comment by lehjr** [comment]
-_-

**Comment by lehjr** [comment]
https://github.com/MachineMuse/MachineMusePowersuits/tree/1.10.2-Java/src/main/java/net/machinemuse/general/gui for one
and used here:
https://github.com/MachineMuse/MachineMusePowersuits/blob/1.10.2-Java/src/main/java/net/machinemuse/powersuits/event/ClientTickHandler.java

**Comment by lehjr** [comment]
Not a trivial change and not one I have time for at the moment. Plus, every time something changes, someone wants the option of having things the old way. Every time you add options, you add code complexity and overhead. And when the mod can no longer run above 1fps, well, it's too late for I told you so.

**Comment by lehjr** [comment]
I'm not knocking the idea. In fact, I like it. It's just right now, I've already got too much on my plate. I would like to get to a beta this month. As it is, I'm spending every spare minute I have doing this, and quite frankly it's not enough. I spent better than half the day Sunday just fixing the meters that were already there and adding another, I really can't spend another day rotating them and changing the layout. Right now, I'm trying to fix the cache for the models so they don't take up gigabytes or more. And guava kinda sucks at figuring out if my damned key already exists in the cache map, so I have to figure out the least cpu intensive way of dealing with that. So if I seem abrasive, well there's why.

**Comment by eyeonus** [comment]
It's fine. I get it. I'm not trying to be a jerk, and I take no offense.

The reason I asked where it is is so I can look at it.

Assuming I can figure out what the code is doing and what I need to change
it, and get a dev build running, I'll take a crack at it myself and then
you can decide whether or not to pull it.

**Comment by lehjr** [comment]
Personally, I like it better than what's there. But I am worried about how it will be received. Still though, playing with it in a dev environment can be enlightening even if you accomplish nothing.

**Comment by lehjr** [comment]
btw, all of the meters are based on the heat meter. so that's the one to start with.

**Comment by eyeonus** [comment]
Yeah, I see that.

That makes thing really easy, since it means only once set of methods needs
altered.

The only think that looks more difficult than straight up math is rotating
the textures to match the rotation of the meters, and I don't think that's
worth the bother. Probably wouldn't even notice on any but the water meter.

**Comment by eyeonus** [comment]
Flipping the glass is a simple matter of flipping the xsize and ysize values and figuring out what the correct xpos and ypos to pass are.

The fluid method is a bit more difficult to fathom, I'll probably be able to figure it out pretty quickly once I have a dev environment running.

**Comment by eyeonus** [comment]
Which version of Forge is this built against right now?

**Comment by lehjr** [comment]
https://github.com/MachineMuse/MachineMusePowersuits/blob/1.10.2-Java/gradle.properties

**Comment by lehjr** [comment]
almost time for the next alpha

**Comment by lehjr** [comment]
rendering on an armor stand is now fixed

**Comment by lehjr** [comment]
Just tested with Bibliocraft armor stands, working there as well.

**Comment by lehjr** [comment]
Compact machines personal shrinking device uses fluid now. So now the module will need a fluid tank and handling code to replicate this instead of breaking the new mechanic. Fun. Guess that will be alpha 12 or later.

**Comment by lehjr** [comment]
Btw, alpha 11 is up (as of Friday).

**Comment by Corruptdead** [comment]
Couple issues I've noticed; The night vision module used the night vision potion effect and has the constant annoying  particle effect, and the axe, shovel, and pick modules are severely underpowered, I've adjusted as much as i can but they mine slightly quicker than by hand but slower than wood tools, however i know they are working because they can break the blocks they're intended for.

**Comment by lehjr** [comment]
The block breaking modules have always been problematic because of the way they handle breaking blocks. I need to revisit the code and do some tweaking, maybe even finish that 3x3 diamond drill.

The potion effects will gone in the next alpha. It was something I meant to do awhile ago and forgot about.

**Comment by lehjr** [comment]
The sprint assist module is actually more broken than I realized. The variable ```horzMovement``` is always 0 unless player teleports;
```double horzMovement = player.distanceWalkedModified - player.prevDistanceWalkedModified;``` would probably be a more sensible alternative.

**Comment by lehjr** [comment]
Updated the list. Not much left for the next alpha.

**Comment by lehjr** [comment]
The ore scanner is getting a rewrite to replace the hard coded ore-value mapping with a list from a file that users can edit. It will probably use a similar setup as the recipes, allowing both registry names and oredict entries.

**Comment by lehjr** [comment]
Ore scanner finally works. Leaf blower is done. Also fixed the hoe module to break plants (like tall grass, flowers, w/e).

**Comment by eyeonus** [comment]
Sweet.

**Comment by lehjr** [comment]
Remaining items will be moved to Beta.

**Comment by eyeonus** [comment]
So if this has made it to the beta stage, maybe you want to correct this on the OP to reflect:
"in the current Alpha build that need to be fixed before the 1.10.2 branch will have beta builds"

*current Beta
*will have release builds

**Comment by eyeonus** [comment]
Also, good job on this.

**Comment by lehjr** [comment]
Well, I did have some help. Between the work you did and the valid bug reports I got and the feedback I got, I was able to get most things fixed. There are probably a few bugs still, but the point of such a long alpha was to get most things working. The elytra flight is still a big milestone on the map. There's a PR for it in Forge, but it seems to have stalled. The challenge is that it is that the flight checks are hard coded, spanning multiple files, and called every tick and checks to see if an Elytra is equipped rather than if the equipped item is an instance of it. Even an instance check would fail anyway, since you cannot extend multiple classes. Both armor and elytra extend Item, but neither is an instance of the other.

**Comment by eyeonus** [comment]
That sounds like a whole lot of bother. Why do we want Elytra flight again? Why is the current jetpack flight not good enough?

**Comment by lehjr** [comment]
mainly for the animation. That's really it.  EnderIO does have a way of dealing with it, so I may be able to do something with it if EnderIO is installed, but honestly, I don't know.

**Comment by eyeonus** [comment]
I suppose copying the relevant code and modifying to suit our purpose is out of the question.

**Comment by lehjr** [comment]
I'd rather not do that. This is part of what EnderIO uses: https://github.com/SleepyTrousers/EnderCore/blob/1.10/src/main/java/com/enderio/core/common/transform/EnderCoreTransformer.java#L224-L262

**Comment by eyeonus** [comment]
Yup, don't understand that code.

It looks like they're loading in the vanilla code and overriding it to add an EnderIO thing to the list of things that will return true?

**Comment by lehjr** [comment]
yeah. The vanilla code is a mess. Waiting to see what forge does with it because anything they do will break everything that overrides all of that.

**Comment by eyeonus** [comment]
We could do the same thing, while waiting for the Forge PR to happen, and fix it to use whatever Forge does after.

Or just hold off until the PR is resolved and decide what to do then.

**Comment by lehjr** [comment]
I'm going to wait for awhile. I'm not about to steal that code and keep in mind that as more people hack into it to make their own code work, the more likely it is to break. Regardless of what means is used to get it working, it requires major changes in the code for the jetpack and glider modules, and probably eliminating the parachute module. The controls are completely different as is the concept of flight. The jetpack would need to be changed to provide thrust in the direction the player is facing, and the glider would need to be changed to accommodate to allow for the jetpack to provide forward thrust. So basically both would have to work together. Then the rendering for the skinned chestplate would have to allow for the elytra wings (which are animated) while the high polly chest plate would need to show the "jetpack" and "wings" (fixed model). It's kind of a can of worms to be honest.

**Comment by eyeonus** [comment]
Yeah. Screw that. Save for after release. :)

**Comment by lehjr** [comment]
That amount of changes would be very unpredictable, so it would be an alpha build. But since there aren't many things left on the list, I'll probably have to address it sooner than I would like. Meanwhile there is a small list left, some of which will take time. Maybe I'll do the wireless grid thing for Refined Storage next. AE2 I'm not going to bother with since it looks like development has stalled completely. However, AE3 might be something to keep an eye out for, as it is being actively worked on. But there are no builds yet, so I have no idea how far along it is.

**Comment by eyeonus** [comment]
As far as that is concerned, I'd still say hold off until the Forge thing is resolved. If that means waiting until after this has a release build, then I'd suggest forking the elytra stuff and releasing alphas based on the stable release as "[mod].[release version].ELYTRA_FLIGHT_ALPHA_[num]".

That way people can have a stable build without elytra flight, and only those that want to test the elytra stuff will "upgrade".

(I am aware that some people will upgrade blindly, so it does mean getting some headaches from stupid people stupiding....)

I think getting the RS wireless grid working would be a great idea.

In the meantime, I'm working on the radial menu. Planning stage right now, partly because work is taking up nearly all my time, but I've got some ideas.

**Comment by lehjr** [comment]
Kinda wondering where 1.11.2 should be on the road map. I don't have any plans for it yet, but I am wondering if I should.

**Comment by eyeonus** [comment]
How different is 1.11.2 to 1.10.2? I would vote to focus on getting a release for 1.10.2 first.

**Comment by lehjr** [comment]
Without really looking, it's hard to say. Release isn't that far off, at least not as far as I can tell. It's hard without a lot of feedback. So far the bug reports are very few, fewest I've seen since I've been around. The only major thing that is left is the module for Refined Storage. There was a module for Compact machines, but the 1.10.2 port adds some type of fluid to the personal shrinking device, and I really don't want to add fluids to the power fist just to deal with it nor do I want to break the mechanics of the mod. Other things on the list are lower priority for release.

**Comment by eyeonus** [comment]
I've been a bit busy, but I'll check it out as soon as I've got time.

**Comment by eyeonus** [comment]
Also, few bug reports is a good sign.

Does compact machines have a fluid battery thing? That is, a battery, but for the fluid? Maybe make the module work if the player has fluid in one of those in the inventory?

**Comment by lehjr** [comment]
I don't think it's all that different, other than ItemStacks can not be null but instead have to be air. It's not that big of a deal anyway. 

Few bug reports is probably because of the low number of downloads, which is I was considering moving to the next version. I might be able to do both for awhile, I dunno. Buildcraft was skipping 1.10.2 and going straight to 1.11.2, so there's that. I was also thinking about doing a port of LWJGL-3 as a drop in replacement for this old legacy version. Official LWJGL3 plans seem to be a ways off and it would be nice to be able to use it now.

**Comment by lehjr** [comment]
I guess I'll update and actually implement GregTech recipes since it was never fully implemented. The GregTech recipes should be extracted if the mod is present, and Industrialcraft recipes should only be extracted if it isn't.

**Comment by lehjr** [comment]
Ok so gregtech recipes are updated and the wireless module for Refined Storage is in the works.

**Comment by lehjr** [comment]
Did get the refined storage module working, not finished but working. But I will have to make sure the code can be adjusted so that the mod is not a hard dependency for the end user. It won't be possible for the mod to be a soft dependency in the development environment though.

**Comment by lehjr** [comment]
Ok, so if this works, the wireless grid module will use either a wireless grid or if the wireless crafting grid is installed, that will be used instead. Just need to set up the power drain and make sure this is still a soft dependency.

**Comment by lehjr** [comment]
A couple ideas I'm looking at is a module for the powerfist using Mekanism's Atomic Disassembler's vein mining capability and a "flight pack" as a sort of jetpack/glider replacement for the elytra flight. This still raises the question of how to achieve this efficiently and without copying someone else's code.

**Comment by lehjr** [comment]
Bumped the last alpha up to a beta. I'll give it a few days and if there aren't any bug reports, I'll set it as a release. Meanwhile, I'm actually spending a few days using the mod to see what it really needs. Next week I'll work on trying to get a module working for Scannable. I did try but my attempts failed. So I need to rethink how I want to try to achieve it.

**Comment by lehjr** [comment]
OK, first release is up. This week I will start looking at 1.11.2.

**Comment by lehjr** [comment]
I have a rough 1.11.2 port done, all but translations not working. However, I'm considering rewriting a few things, like power handling and implementing Tesla and Forge power support. RF needs an update too. It's time to start looking at capabilities more seriously. It's time to start looking at ways to improve performance.
